### PR TITLE
feat(recovery-key): Create initial components for inline recovery key setup flow

### PR DESCRIPTION
**Pull request by LZoog**
Because:
* We want to take Sync users to an 'inline' (after login) recovery key setup flow if they don't already have a key

This commit:
* Creates initial components for this flow and adds basic tests and l10n
* Contains some refactoring to share with the download component with Settings

closes FXA-10078

---

The SVG in the PDF is going to need to be a separate ticket, I spun my wheels on it for a bit and have an idea/WIP branch but pivoted to get this one done since it unblocks other work in the epic.

### CODE REVIEW

File: packages/fxa-settings/src/components/InlineRecoveryKeySetupCreate/index.test.tsx
Comment (vbudhram) [inline packages/fxa-settings/src/components/InlineRecoveryKeySetupCreate/index.test.tsx:17]: The figma doc isn't clear what this page should be, it looks this this currently

Screenshot 2024-09-06 at 1 46 33 PM

File: packages/fxa-settings/src/components/InlineRecoveryKeySetupCreate/index.tsx
Comment (vbudhram) [inline packages/fxa-settings/src/components/InlineRecoveryKeySetupCreate/index.tsx:19]: Looks like this also need a banner component "You're signed in to Firefox"

**Review by vbudhram — COMMENTED** [review]
@LZoog In the figma there isn't a progress bar indicator, not sure if that is needed to be included. I was able to look at the storybook components and they seem to match the context in the figma diagrams.

**Comment by LZoog** [comment]
@vbudhram I was following the [Figma designs here](https://www.figma.com/design/v5d5xDL7R2xYfN3Hx9gtN2/Account-Recovery-Key---Continuous-onboarding?node-id=2806-44522&node-type=INSTANCE&t=vdg99Q2akjN0kchJ-0), but now I'm confused because I see the other section you're talking about so... I'll check with Eduardo.

**Review by vbudhram — APPROVED** [review]
@LZoog Thanks! This matches the mock ups :shipit:

File: packages/fxa-settings/src/components/InlineRecoveryKeySetupCreate/en.ftl
Comment (bcolsson) [inline packages/fxa-settings/src/components/InlineRecoveryKeySetupCreate/en.ftl:5]:
```suggestion
# This is a subheader asking users to create an account recovery key, indicating it will only take a moment to complete. 
# Possible alternatives: "Protect your data. It will only take a moment."
inline-recovery-key-setup-create-subheader = Got a minute to protect your data?
```
"Got a minute" is idiomatic and a bit colloquial which might make it hard for some languages to offer a natural localization here. It'd be nice to add a comment here will a more simplified alternative to assist localizers. (Suggestion above is just a suggestion, please feel free to adjust.)

**Review by bcolsson — APPROVED** [review]
Approving so a second review isn't required but please see my review comment about adding a comment to `inline-recovery-key-setup-create-subheader`.

Also, `RecoveryKeySetupDownload/en.ftl` is empty - is this missing strings?

File: packages/fxa-settings/src/components/InlineRecoveryKeySetupCreate/en.ftl
Comment (LZoog) [reply packages/fxa-settings/src/components/InlineRecoveryKeySetupCreate/en.ftl:5]: Nice catches on both. I've added that comment and for the missing strings, I meant to shuffle them over from a different component, so that's done. 👍

**Pull request merged**
